src/components/package-show/package-show.js

```diff
@@ -1,137 +1,199 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
 import Icon from '@folio/stripes-components/lib/Icon';
 import PaneHeader from '@folio/stripes-components/lib/PaneHeader';
 import PaneMenu from '@folio/stripes-components/lib/PaneMenu';
+import Button from '@folio/stripes-components/lib/Button';
 
 import Link from '../link';
 import KeyValueLabel from '../key-value-label';
 import List from '../list';
 import TitleListItem from '../title-list-item';
 import ToggleSwitch from '../toggle-switch';
+import Modal from '../modal';
 import { formatISODateWithoutTime } from '../utilities';
 import styles from './package-show.css';
 
-export default function PackageShow({ model, toggleSelected }, { intl, router, queryParams }) {
-  let historyState = router.history.location.state;
-
-  return (
-    <div>
-      {!queryParams.searchType && (
-        <PaneHeader
-          firstMenu={historyState && historyState.eholdings && (
-            <PaneMenu>
-              <button data-test-eholdings-package-details-back-button onClick={() => router.history.goBack()}><Icon icon="left-arrow" /></button>
-            </PaneMenu>
-          )}
-        />
-      )}
-      <div className={styles['detail-container']} data-test-eholdings-package-details>
-        {model.isLoaded ? (
-          <div>
-            <div className={styles['detail-container-header']}>
-              <KeyValueLabel label="Package">
-                <h1 data-test-eholdings-package-details-name>
-                  {model.name}
-                </h1>
-              </KeyValueLabel>
-            </div>
-
-            <KeyValueLabel label="Vendor">
-              <div data-test-eholdings-package-details-vendor>
-                <Link to={`/eholdings/vendors/${model.vendorId}`}>{model.vendorName}</Link>
+export default class PackageShow extends Component {
+  static propTypes = {
+    model: PropTypes.object.isRequired,
+    toggleSelected: PropTypes.func.isRequired
+  };
+
+  static contextTypes = {
+    intl: PropTypes.object,
+    router: PropTypes.object,
+    queryParams: PropTypes.object
+  };
+
+  state = {
+    showSelectionModal: false,
+    packageSelected: this.props.model.isSelected
+  };
+
+  componentWillReceiveProps({ model }) {
+    if (!model.isSaving) {
+      this.setState({ packageSelected: model.isSelected });
+    }
+  }
+
+  handleSelectionToggle = () => {
+    this.setState({ packageSelected: !this.props.model.isSelected });
+    if (this.props.model.isSelected) {
+      this.setState({ showSelectionModal: true });
+    } else {
+      this.props.toggleSelected();
+    }
+  };
+
+  commitSelectionToggle = () => {
+    this.setState({ showSelectionModal: false });
+    this.props.toggleSelected();
+  };
+
+  cancelSelectionToggle = () => {
+    this.setState({ showSelectionModal: false });
+    this.setState({ packageSelected: this.props.model.isSelected });
```

Comment:
These two `setState` should be combined into one:

```javascript
this.setState({
  showSelectionModal: false,
  packageSelected: this.props.model.isSelected
});
```

Reply:
Fixed.